cargo-pgx/src/commands/init.rs

```diff
@@ -31,7 +31,15 @@ static PROCESS_ENV_DENYLIST: &'static [&'static str] = &[
 ];
 
 pub(crate) fn init_pgx(pgx: &Pgx) -> std::result::Result<(), std::io::Error> {
-    let dir = Pgx::home()?;
+    let pgx_dir = Pgx::home()?;
+
+    // We use a tempdir to build the postgres versions in case the user has an existing pgx home
+    // this way, if they cancel the process (say it was an accident) we don't ruin their postgres builds.
+    let temp_dir_path = {
+        let mut temp_dir_path = pgx_dir.clone();
+        temp_dir_path.push("scratch");
+        temp_dir_path
```

Comment:
if the `scactch/` directory already exists, lets go ahead and delete its contents here so that a run after a previous cancel starts with fresh stuff.